lib/ui/compositing.dart

```diff
@@ -51,34 +158,74 @@ class SceneBuilder extends NativeFieldWrapperClass2 {
   SceneBuilder() { _constructor(); }
   void _constructor() native 'SceneBuilder_constructor';
 
+  // Layers used in this scene.
+  //
+  // The key is the layer used. The value is the description of what the layer
+  // is used for, e.g. "pushOpacity" or "addRetained".
+  Map<EngineLayer, String> _usedLayers = <EngineLayer, String>{};
+
+  // In debug mode checks that the `layer` is only used once in a given scene.
+  bool _debugCheckUsedOnce(EngineLayer layer, String usage) {
+    if (layer == null) {
+      return true;
+    }
+
+    assert(
+      !_usedLayers.containsKey(layer),
+      'Layer ${layer.runtimeType} already used.\n'
+      'The layer is already being used for ${_usedLayers[layer]} in this scene.\n'
+      'A layer may only be used once in a given scene.'
+    );
+
+    _usedLayers[layer] = usage;
```

Comment:
this whole method should be in an assert, to make sure we really don't use it in release

Reply:
Done.